libs/common/src/sql-db-entities/fee-record/fee-record.entity.ts

```diff
@@ -181,21 +213,65 @@ export class FeeRecordEntity extends AuditableBaseEntity {
     return feesPaidToUkefForThePeriodAsBig.div(paymentExchangeRateAsBig).round(precision).toNumber();
   }
 
+  /**
+   * Updates the fee record with a status
+   * @param param - The update parameters
+   * @param param.status - The fee record status to set
+   * @param param.requestSource - The request source making the update
+   */
   public updateWithStatus({ status, requestSource }: UpdateWithStatusParams): void {
     this.status = status;
     this.updateLastUpdatedBy(requestSource);
   }
 
+  /**
+   * Updates the fee record with keying data
+   * @param param - The update parameters
+   * @param param.fixedFeeAdjustment - The fixed fee adjustment
+   * @param param.principalBalanceAdjustment - The principal balance adjustment
+   * @param param.status - The status
+   * @param param.requestSource - The request source making the update
+   */
   public updateWithKeyingData({ fixedFeeAdjustment, principalBalanceAdjustment, status, requestSource }: UpdateWithKeyingDataParams): void {
     this.status = status;
     this.fixedFeeAdjustment = fixedFeeAdjustment;
     this.principalBalanceAdjustment = principalBalanceAdjustment;
     this.updateLastUpdatedBy(requestSource);
   }
 
+  /**
+   * Removes all payments from the fee record
+   * @param param - The update parameters
+   * @param param.requestSource - The request source making the update
+   */
```

Comment:
For my own understanding, why would we want to remove all payments?

Reply:
This is one of the features we need to support, but the real purpose of this method is to communicate that when you delete all payments from a fee record, you must also update the status of the fee record to `'TO_DO'`

Reply:
OK thanks - what is `TO_DO`? I keep seeing this. Is this something we need to do, or is this a "business/design status"?

Reply:
`TO_DO` is one of the possible fee record statuses